lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -21,11 +47,17 @@ def initialize(serializer, options = {})
         def serializable_hash(options = nil)
           options ||= {}
 
-          if serializer.respond_to?(:each)
-            serializable_hash_for_collection(options)
-          else
-            serializable_hash_for_single_resource(options)
-          end
+          hash =
+            case
```

Comment:
I feel quite strongly about it. I personally think we should leverage the functional constructs of ruby in order to have the clearest/"no surprise-est" code possible, and I think using the return value of an `if/else` allows for that.